rotkehlchen/tests/api/test_locations.py

```diff
@@ -0,0 +1,58 @@
+import os
+from pathlib import Path
+
+import pytest
+import requests
+
+from rotkehlchen.chain.ethereum.typing import string_to_ethereum_address
+from rotkehlchen.tests.utils.api import (
+    api_url_for,
+    assert_proper_response_with_result,
+)
+from rotkehlchen.tests.utils.exchanges import mock_exchange_data_in_db
+from rotkehlchen.typing import Location
+
+UNISWAP_ADDR = string_to_ethereum_address('0xcaf012cB72f2c7152b255E091837E3a628F739e7')
+
+
+@pytest.mark.parametrize('added_exchanges', [(Location.BINANCE, Location.POLONIEX)])
+@pytest.mark.parametrize('ethereum_accounts', [[UNISWAP_ADDR]])
+@pytest.mark.parametrize('ethereum_modules', [['uniswap']])
+@pytest.mark.parametrize('start_with_valid_premium', [True])
+def test_get_associated_locations(
+    rotkehlchen_api_server_with_exchanges,
+    added_exchanges,
+    ethereum_accounts,  # pylint: disable=unused-argument
+    start_with_valid_premium,  # pylint: disable=unused-argument
+):
+    rotki = rotkehlchen_api_server_with_exchanges.rest_api.rotkehlchen
+    mock_exchange_data_in_db(added_exchanges, rotki)
+
+    # Import from nexo
+    dir_path = Path(os.path.dirname(os.path.dirname(os.path.realpath(__file__))))
+    filepath = dir_path / 'data' / 'nexo.csv'
+
+    json_data = {'source': 'nexo', 'file': str(filepath)}
+    requests.put(
+        api_url_for(
+            rotkehlchen_api_server_with_exchanges,
+            'dataimportresource',
+        ), json=json_data,
+    )
+
+    # Query uniswap trades
+    requests.get(api_url_for(
+        rotkehlchen_api_server_with_exchanges, 'uniswaptradeshistoryresource'),
+        json={'async_query': False},
+    )
```

Comment:
This is an api test that should test the the api's associated location endpoint works fine. Do not call on the graph to populate uniswap. Populate it via mocking. Just add some stuff in the DB.

Our test should be as simple and as fast as possible. This test does not need to query anything outside of rotki and thus it MUST NOT.